### DISCUSSION TITLE
Virtual Images error installing, port 8500 connection refused.

### DISCUSSION THREAD

**Discussion by srmcdowell** [Seagate/cortx#379] [General] [answered]
Hi,  

Following the installation instructions for the VM image, I'm encountering a number of errors.   Whatever you're expecting to be on port 8500 seems absent (I've checked firewall rules + looked for a listen on that port). 

All services described in the installation troubleshooting section are up, but the "cluster" is down. 

I've had the same results with both of the virtual image packages. 

Command to reproduce (as root): 
# sh /opt/seagate/cortx/provisioner/cli/virtual_appliance/bootstrap.sh 

Output: 

INFO: Configuring haproxy
[0;31msrvnode-1:[0;0m
 [0;32m Name: rsyslog - Function: pkg.installed - Result: Clean Started: - 13:58:12.389607 Duration: 2688.358 ms[0;0m
 [0;32m Name: rsyslog.service - Function: service.running - Result: Clean Started: - 13:58:15.101385 Duration: 62.701 ms[0;0m
 [0;32m Name: certs - Function: group.present - Result: Clean Started: - 13:58:15.177862 Duration: 1.873 ms[0;0m
[0;31m----------[0;0m
    [0;31m      ID: Setup HAProxy config[0;0m
    [0;31mFunction: file.managed[0;0m
    [0;31m    Name: /etc/haproxy/haproxy.cfg[0;0m
    [0;31m  Result: False[0;0m
    [0;31m Comment: Unable to manage file: Jinja variable 'dict object' has no attribute 'ens256'[0;0m
    [0;31m Started: 13:58:15.193038[0;0m
    [0;31mDuration: 254.003 ms[0;0m
[0;31m     Changes:   [0;0m
 [0;32m Name: /etc/haproxy/errors/503.http - Function: file.managed - Result: Clean Started: - 13:58:15.447225 Duration: 26.824 ms[0;0m
 [0;32m Name: /etc/logrotate.d/haproxy - Function: file.managed - Result: Clean Started: - 13:58:15.474203 Duration: 17.057 ms[0;0m
 [0;32m Name: /etc/cron.hourly/logrotate - Function: file.managed - Result: Clean Started: - 13:58:15.491434 Duration: 17.576 ms[0;0m
 [0;32m Name: /etc/rsyslog.d/haproxy.conf - Function: file.managed - Result: Clean Started: - 13:58:15.509170 Duration: 16.194 ms[0;0m
[0;36m
Summary for srvnode-1
------------[0;0m
[0;32mSucceeded: 7[0;0m
[0;31mFailed:    1[0;0m
[0;36m------------
Total states run:     8[0;0m
[0;36mTotal run time:   3.085 s[0;0m
INFO: Configuring s3 ips in /etc/hosts
[0;1;31msrvnode-1:[0;0m
    [0;1;31mData failed to compile:[0;0m
[0;1;31m----------
    Rendering SLS 'base:components.s3server.config_hosts' failed: Jinja variable 'dict object' has no attribute 'ens224'[0;0m
INFO: Restarting haproxy
INFO: Restarting slapd
INFO: Restarting rabbitmq-server
INFO: Restarting elasticsearch
INFO: Restarting s3authserver
INFO: Bootstraping cluster
2020-09-28 13:58:21: Generating cluster configuration...INFO: Configuring sspl
[0;31msrvnode-1:[0;0m
[0;31m----------[0;0m
    [0;31m      ID: Add common config - system information to Consul[0;0m
    [0;31mFunction: cmd.run[0;0m
    [0;31m    Name: /opt/seagate/cortx/hare/bin/consul kv put system_information/operating_system "$(cat /etc/system-release)"
/opt/seagate/cortx/hare/bin/consul kv put system_information/kernel_version 3.10.0-1062.el7.x86_64
/opt/seagate/cortx/hare/bin/consul kv put system_information/product single
/opt/seagate/cortx/hare/bin/consul kv put system_information/site_id 001
/opt/seagate/cortx/hare/bin/consul kv put system_information/rack_id 001
/opt/seagate/cortx/hare/bin/consul kv put system_information/cluster_id f639e0da-f1b6-4c32-a3c0-86e820beaafa
/opt/seagate/cortx/hare/bin/consul kv put system_information/srvnode-1/node_id 55876568-ad42-4666-b573-e42616577528
/opt/seagate/cortx/hare/bin/consul kv put system_information/syslog_host localhost
/opt/seagate/cortx/hare/bin/consul kv put system_information/syslog_port 514
/opt/seagate/cortx/hare/bin/consul kv put system_information/healthmappath /opt/seagate/cortx/provisioner/generated_configs/healthmap//ees-schema.json
[0;0m
    [0;31m  Result: False[0;0m
    [0;31m Comment: Command "/opt/seagate/cortx/hare/bin/consul kv put system_information/operating_system "$(cat /etc/system-release)"
              /opt/seagate/cortx/hare/bin/consul kv put system_information/kernel_version 3.10.0-1062.el7.x86_64
              /opt/seagate/cortx/hare/bin/consul kv put system_information/product single
              /opt/seagate/cortx/hare/bin/consul kv put system_information/site_id 001
              /opt/seagate/cortx/hare/bin/consul kv put system_information/rack_id 001
              /opt/seagate/cortx/hare/bin/consul kv put system_information/cluster_id f639e0da-f1b6-4c32-a3c0-86e820beaafa
              /opt/seagate/cortx/hare/bin/consul kv put system_information/srvnode-1/node_id 55876568-ad42-4666-b573-e42616577528
              /opt/seagate/cortx/hare/bin/consul kv put system_information/syslog_host localhost
              /opt/seagate/cortx/hare/bin/consul kv put system_information/syslog_port 514
              /opt/seagate/cortx/hare/bin/consul kv put system_information/healthmappath /opt/seagate/cortx/provisioner/generated_configs/healthmap//ees-schema.json
              " run[0;0m
    [0;31m Started: 13:58:26.474823[0;0m
    [0;31mDuration: 1364.357 ms[0;0m
[0;31m     Changes:   
              [0;36m----------[0;0m
              [0;36mpid[0;0m:
                  [0;1;33m5073[0;0m
              [0;36mretcode[0;0m:
                  [0;1;33m1[0;0m
              [0;36mstderr[0;0m:
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/operating_system: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/kernel_version: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/product: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/site_id: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/rack_id: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/cluster_id: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/srvnode-1/node_id: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/syslog_host: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/syslog_port: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/system_information/healthmappath: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
              [0;36mstdout[0;0m:[0;0m
[0;31m----------[0;0m
    [0;31m      ID: Add common config - rabbitmq cluster to Consul[0;0m
    [0;31mFunction: cmd.run[0;0m
    [0;31m    Name: /opt/seagate/cortx/hare/bin/consul kv put rabbitmq/cluster_nodes localhost
/opt/seagate/cortx/hare/bin/consul kv put rabbitmq/erlang_cookie QLDZYPYEYGHECTHYQXFJ
[0;0m
    [0;31m  Result: False[0;0m
    [0;31m Comment: Command "/opt/seagate/cortx/hare/bin/consul kv put rabbitmq/cluster_nodes localhost
              /opt/seagate/cortx/hare/bin/consul kv put rabbitmq/erlang_cookie QLDZYPYEYGHECTHYQXFJ
              " run[0;0m
    [0;31m Started: 13:58:27.839480[0;0m
    [0;31mDuration: 119.03 ms[0;0m
[0;31m     Changes:   
              [0;36m----------[0;0m
              [0;36mpid[0;0m:
                  [0;1;33m5183[0;0m
              [0;36mretcode[0;0m:
                  [0;1;33m1[0;0m
              [0;36mstderr[0;0m:
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/rabbitmq/cluster_nodes: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/rabbitmq/erlang_cookie: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
              [0;36mstdout[0;0m:[0;0m
[0;31m----------[0;0m
    [0;31m      ID: Add common config - BMC to Consul[0;0m
    [0;31mFunction: cmd.run[0;0m
    [0;31m    Name: 
/opt/seagate/cortx/hare/bin/consul kv put bmc/srvnode-1/ip None
/opt/seagate/cortx/hare/bin/consul kv put bmc/srvnode-1/user ADMIN
/opt/seagate/cortx/hare/bin/consul kv put bmc/srvnode-1/secret gAAAAABfX4IXLfzCmB8jCkF4dLFil5557ow-Kwq0ZzqQGN0TTAvYPvGpBhSRTFJD5Ov7Anl6gFK7vErgXZg9btJxB1OBqnZ-MA==
[0;0m
    [0;31m  Result: False[0;0m
    [0;31m Comment: Command "
              /opt/seagate/cortx/hare/bin/consul kv put bmc/srvnode-1/ip None
              /opt/seagate/cortx/hare/bin/consul kv put bmc/srvnode-1/user ADMIN
              /opt/seagate/cortx/hare/bin/consul kv put bmc/srvnode-1/secret gAAAAABfX4IXLfzCmB8jCkF4dLFil5557ow-Kwq0ZzqQGN0TTAvYPvGpBhSRTFJD5Ov7Anl6gFK7vErgXZg9btJxB1OBqnZ-MA==
              " run[0;0m
    [0;31m Started: 13:58:27.958847[0;0m
    [0;31mDuration: 192.776 ms[0;0m
[0;31m     Changes:   
              [0;36m----------[0;0m
              [0;36mpid[0;0m:
                  [0;1;33m5206[0;0m
              [0;36mretcode[0;0m:
                  [0;1;33m1[0;0m
              [0;36mstderr[0;0m:
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/bmc/srvnode-1/ip: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/bmc/srvnode-1/user: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/bmc/srvnode-1/secret: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
              [0;36mstdout[0;0m:[0;0m
[0;31m----------[0;0m
    [0;31m      ID: Add common config - storage enclosure to Consul[0;0m
    [0;31mFunction: cmd.run[0;0m
    [0;31m    Name: /opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/primary_mc/ip 10.0.0.2
/opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/primary_mc/port 80
/opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/secondary_mc/ip 10.0.0.3
/opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/secondary_mc/port 80
/opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/user manage
/opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/password gAAAAABfX4IXBO36r5czFIKxHiq0yDtC5qhWxHwLOkgv7yiZSJ9a9JljMquC2-aBCB-rNCZvZeF86FDIVBe80PQzg5kxLiKKEA==
[0;0m
    [0;31m  Result: False[0;0m
    [0;31m Comment: Command "/opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/primary_mc/ip 10.0.0.2
              /opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/primary_mc/port 80
              /opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/secondary_mc/ip 10.0.0.3
              /opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/secondary_mc/port 80
              /opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/user manage
              /opt/seagate/cortx/hare/bin/consul kv put storage_enclosure/controller/password gAAAAABfX4IXBO36r5czFIKxHiq0yDtC5qhWxHwLOkgv7yiZSJ9a9JljMquC2-aBCB-rNCZvZeF86FDIVBe80PQzg5kxLiKKEA==
              " run[0;0m
    [0;31m Started: 13:58:28.151868[0;0m
    [0;31mDuration: 282.95 ms[0;0m
[0;31m     Changes:   
              [0;36m----------[0;0m
              [0;36mpid[0;0m:
                  [0;1;33m5241[0;0m
              [0;36mretcode[0;0m:
                  [0;1;33m1[0;0m
              [0;36mstderr[0;0m:
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/storage_enclosure/controller/primary_mc/ip: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/storage_enclosure/controller/primary_mc/port: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/storage_enclosure/controller/secondary_mc/ip: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/storage_enclosure/controller/secondary_mc/port: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/storage_enclosure/controller/user: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
                  [0;32mError! Failed writing data: Put http://127.0.0.1:8500/v1/kv/storage_enclosure/controller/password: dial tcp 127.0.0.1:8500: connect: connection refused[0;0m
              [0;36mstdout[0;0m:[0;0m
[0;36m

**Comment by johnbent** [comment]
Thank you for reporting this @srmcdowell and I apologize for the difficulties!  What hypervisor are you using?

**Comment by d283w** [comment]
I have similar issues on ESX 7.0.  I think one of the issues is that interfaces ens224 and ens256 assume you have DHCP provisioning IP addresses.  I was able to get further by manually editing ifcfg-ens224 and ifcfg-ens256 to assign IPs manually prior to running bootstrap.sh.  I can get further and the mgmt UI is now responding, but I can't get S3 to work.

**Comment by justinzw** [comment]
thanks @d283w for mentioning this.

@srmcdowell I'm wondering whether you can run `ip a l` and check to see that all interfaces have an IP address associated to it.

For reference -> Here is my output when I run it.

![image](https://user-images.githubusercontent.com/950172/94479059-10447000-0189-11eb-867d-e1d257c020aa.png)

**Comment by d283w** [comment, marked as answer]
Something's going wrong with the process to modify haproxy.cfg and /etc/hosts

[0;31m----------[0;0m
    [0;31m      ID: Setup HAProxy config[0;0m
    [0;31mFunction: file.managed[0;0m
    [0;31m    Name: /etc/haproxy/haproxy.cfg[0;0m
    [0;31m  Result: False[0;0m
    [0;31m Comment: Unable to manage file: Jinja variable list object has no element 0[0;0m
    [0;31m Started: 18:09:57.376345[0;0m
    [0;31mDuration: 177.684 ms[0;0m
[0;31m     Changes:   [0;0m

INFO: Configuring s3 ips in /etc/hosts
[0;1;31msrvnode-1:[0;0m
    [0;1;31mData failed to compile:[0;0m
[0;1;31m----------
    Rendering SLS 'base:components.s3server.config_hosts' failed: Jinja variable list object has no element 0[0;0m

Manually modifying both files to remove the 192.168.X.X references, replacing with the IP for the ens256 interface finally resulted in success for the s3-sanity-test.

**Comment by johnbent** [comment]
@srmcdowell , can you please try the instructions from @d283w and @justinzw to see if that works for you?

**Reply by srmcdowell** [reply]
These instructions seems have done the trick -- thanks! 

I'd maintain that this is an issue, since the VMs aren't usable without corrective action (that is only documented here) is required by the end-user.

**Comment by johnbent** [comment]
@justinzw , is this a Discussion or an Issue?

**Reply by justinzw** [reply]
yah I agree with @srmcdowell  that this is an issue.

**Reply by johnbent** [reply]
Can we transfer it to an Issue?  I couldn’t see how....

**Reply by Saumya-Sunder** [reply]
created issue https://github.com/Seagate/cortx/issues/580
